### DISCUSSION TITLE
LDAP configuration

### DISCUSSION THREAD

**Discussion by Adel-Akbi** [tchiotludo/akhq#996] [Q&A]
Hello,

We are trying to configure the LDAP in order to allow users to connect using their ID.. we tried before this since we were using the version 0.13 but now we recently upgraded to 0.19 telling that the version 0.19 bring the solution of our last problem.. but we still have the same issue. 

We have this error message : 
```
Jan 11 17:38:37 SERVER akhq: at java.base/java.lang.Thread.run(Thread.java:829)
Jan 11 17:38:37 SERVER akhq: 2022-01-11 17:38:37,775 #033[31mWARN #033[0;39m #033[35mGroup-1-14#033[0;39m #033[36mtractChannelHandlerContext#033[0;39m An exception 'io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [enabled] of method [setEnabled] of class: io.micronaut.configuration.security.ldap.configuration.LdapConfiguration
Jan 11 17:38:37 SERVER akhq: Message: Error resolving property value [micronaut.security.ldap.default.enabled]. Property doesn't exist
Jan 11 17:38:37 SERVER akhq: Path Taken: new SecurityFilter(Collection securityRules,[Collection authenticationFetchers],SecurityConfiguration securityConfiguration) --> new BasicAuthAuthenticationFetcher([Authenticator authenticator],TokenConfiguration configuration) --> new Authenticator([Collection authenticationProviders],SecurityConfiguration securityConfiguration) --> LdapConfiguration.setEnabled([boolean enabled])' [enable DEBUG level for full stacktrace] was thrown by a user handler's exceptionCaught() method while handling the following exception:
Jan 11 17:38:37 SERVER akhq: io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [enabled] of method [setEnabled] of class: io.micronaut.configuration.security.ldap.configuration.LdapConfiguration
Jan 11 17:38:37 SERVER akhq: Message: Error resolving property value [micronaut.security.ldap.default.enabled]. Property doesn't exist
Jan 11 17:38:37 SERVER akhq: Path Taken: new SecurityFilter(Collection securityRules,[Collection authenticationFetchers],SecurityConfiguration securityConfiguration) --> new BasicAuthAuthenticationFetcher([Authenticator authenticator],TokenConfiguration configuration) --> new Authenticator([Collection authenticationProviders],SecurityConfiguration securityConfiguration) --> LdapConfiguration.setEnabled([boolean enabled])
```
Since we are a bank company, we are not allowed to display the content of application.yml, because of this, is it possible to schedule a teams call and try to fix the problem ? we tried all the possible solutions but no way...

Regards,

**Comment by tchiotludo** [comment]
```yaml 
micronaut:
  security: 
    ldap: 
      default: 
        enabled: true
```
since that we have missing `enabled: true` on your yaml configuration ?

**Comment by Adel-Akbi** [comment]
it's already  set

**Comment by tchiotludo** [comment]
We can send an anonymized yaml please ? remove all the mention from your company ;)

**Comment by Adel-Akbi** [comment]
```
micronaut:
  security:
    enabled: true
    ldap:
      default:
        enabled: true
        context:
        server: 'ldap://SERVER:389'
        managerDn: 'cn=USER,OU=Service,DC=dom******,DC=***'
        managerPassword: '****'
      search:
        base: "OU=***,OU=***,DC=dom*******,DC=**"
      groups:
        enabled: true
        base: "OU=**,OU=**,DC=dom*****,DC=**"
        #filter: "member={0}"

```
    ldap:
      groups:
       - name: GROUPENAME
         groups: # Akhq groups list
           - admin

I've fixed the previous error message, but now I have this : 

] [Duration: 0 ms] [Url: GET /ui/static/js/main.d8ca4bec.chunk.js] [Status: 200] [Ip: /@IP] [User: Anonymous]
Jan 12 11:12:20 SERVER akhq: 2022-01-12 11:12:20,131 #033[34mINFO #033[0;39m #033[35mGroup-1-11#033[0;39m #033[36morg.akhq.log.access       #033[0;39m [Date: 2022-01-12T11:12:20.131195+01:00] [Duration: 0 ms] [Url: GET /ui/favicon.ico] [Status: 200] [Ip: /@IP] [User: Anonymous]
Jan 12 11:12:20 SERVER akhq: 2022-01-12 11:12:20,148 #033[34mINFO #033[0;39m #033[35mr-thread-2#033[0;39m #033[36morg.akhq.log.access       #033[0;39m [Date: 2022-01-12T11:12:20.128329+01:00] [Duration: 19 ms] [Url: GET /api/auths] [Status: 200] [Ip: /@IP] [User: Anonymous]
Jan 12 11:12:20 SERVER akhq: 2022-01-12 11:12:20,157 #033[34mINFO #033[0;39m #033[35mr-thread-1#033[0;39m #033[36morg.akhq.log.access       #033[0;39m [Date: 2022-01-12T11:12:20.128346+01:00] [Duration: 25 ms] [Url: GET /api/me] [Status: 200] [Ip: /@IP] [User: Anonymous]
Jan 12 11:12:20 SERVER akhq: 2022-01-12 11:12:20,231 #033[34mINFO #033[0;39m #033[35mpGroup-1-9#033[0;39m #033[36morg.akhq.log.access       #033[0;39m [Date: 2022-01-12T11:12:20.231096+01:00] [Duration: 0 ms] [Url: GET /ui/static/media/icon.648ce9c8.svg] [Status: 200] [Ip: /@IP] [User: Anonymous]
Jan 12 11:12:20 SERVER akhq: 2022-01-12 11:12:20,232 #033[34mINFO #033[0;39m #033[35mGroup-1-10#033[0;39m #033[36morg.akhq.log.access       #033[0;39m [Date: 2022-01-12T11:12:20.2326+01:00] [Duration: 0 ms] [Url: GET /ui/static/media/logo.45903e1f.svg] [Status: 200] [Ip: /@IP] [User: Anonymous]

Thanks

**Comment by tchiotludo** [comment]
Please enabled more logs with [this](https://akhq.io/docs/debug.html#debugging-authentication)

**Comment by Adel-Akbi** [comment]
Hello,

I have already executed the debug commands you sent, the problem still persists and no concrete logs... :/

**Comment by tchiotludo** [comment]
does curl return a response code 200 ?

**Comment by Adel-Akbi** [comment]
Yes 
```
# curl -i -X POST -H "Content-Type: application/json"        -d '{ "configuredLevel": "TRACE" }'        http://localhost:28081/loggers/org.akhq.configs
HTTP/1.1 200 OK
Content-Length: 0
date: Thu, 13 Jan 2022 13:35:48 GMT
connection: keep-alive
```

**Comment by tchiotludo** [comment]
Ok good, now try to login and you will have more log on the webserver
